osu.Game/Overlays/DashboardOverlay.cs

```diff
@@ -0,0 +1,179 @@
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    ScrollbarVisible = false,
+                    Child = new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Y,
+                        RelativeSizeAxes = Axes.X,
+                        Direction = FillDirection.Vertical,
+                        Children = new Drawable[]
+                        {
+                            header = new DashboardOverlayHeader
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Depth = -float.MaxValue
+                            },
+                            content = new Container
+                            {
+                                RelativeSizeAxes = Axes.X,
+                                AutoSizeAxes = Axes.Y
+                            }
+                        }
+                    }
+                },
+                loading = new LoadingLayer(content),
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            background.Colour = ColourProvider.Background5;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            localUser.BindTo(api.LocalUser);
+            localUser.BindValueChanged(_ => onTabChanged());
+            header.Current.BindValueChanged(_ => onTabChanged());
+        }
+
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            if (!content.Any())
+                header.Current.TriggerChange();
+        }
+
+        protected override void PopOutComplete()
+        {
+            base.PopOutComplete();
+            loadDisplay(null);
+        }
+
+        private void onTabChanged()
+        {
+            lastRequest?.Cancel();
+            cancellationToken?.Cancel();
+
+            loading.Show();
+
+            // We may want to use OnlineViewContainer after https://github.com/ppy/osu/pull/8044 merge
+            if (!api.IsLoggedIn)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            var request = createScopedRequest();
+            lastRequest = request;
+
+            if (request == null)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            request.Success += () => Schedule(() => loadDisplay(createDisplayFromResponse(request)));
+            request.Failure += _ => Schedule(() => loadDisplay(null));
+
+            api.Queue(request);
+        }
+
+        private void loadDisplay(Drawable display)
+        {
+            scrollFlow.ScrollToStart();
+
+            if (display == null)
+            {
+                content.Clear();
+                loading.Hide();
+                return;
+            }
```

Comment:
As above - if `null` was disallowed here and `Empty` was passed instead this logic could go away.

Reply:
I need this logic though. I need to check if we have any content in `PopIn()`, so we don't create new requests if spamming social button in toolbar

Reply:
Sounds like something `BindValueChanged(..., true)` could do.

Reply:
I added a simple bool flag to check if overlay has been fully closed before sending a new request.